app/connectors/AuthProviderClient.scala

```diff
@@ -47,7 +49,9 @@ trait AuthProviderClient {
 
   val allRoles = List(FaststreamTeamRole, ServiceSupportRole, ServiceAdminRole, SuperAdminRole, TechnicalAdminRole)
 
-  def getRole(roleName: String) = allRoles.find(_.name == roleName).getOrElse(throw new UserRoleDoesNotExist(s"No such role: $roleName"))
+  def getRole(roleName: String) = allRoles.find(_.name == roleName).getOrElse(
```

Comment:
Wouldn't be better to use a Try as returned type and let the client to deal with the possible exception?
Perhaps out of scope but it's a more functional way to deal with Exception in general.